pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -4,67 +4,500 @@
+		return err
+	}
+
+	if err := writeCommand(builder, container.Command, indent+indentUsing); err != nil {
+		return err
+	}
+
+	if err := writeEnvRules(builder, container.EnvRules, indent+indentUsing); err != nil {
+		return err
+	}
+
+	if err := writeLayers(builder, container.Layers, indent+indentUsing); err != nil {
+		return err
+	}
+
+	if err := writeMounts(builder, container.Mounts, indent+indentUsing); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s\"allow_elevated\": %v,\n", indent+indentUsing, container.AllowElevated)); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s\"working_dir\": \"%s\"\n", indent+indentUsing, container.WorkingDir)); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s}", indent)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addContainers(builder *strings.Builder, containers []*securityPolicyContainer) error {
+	if _, err := builder.WriteString("containers := [\n"); err != nil {
+		return err
+	}
+
+	for i, container := range containers {
+		if err := writeContainer(builder, container, indentUsing); err != nil {
+			return err
+		}
+
+		var end string
+		if i < len(containers)-1 {
+			end = ",\n"
+		} else {
+			end = "\n"
+		}
+
+		if _, err := builder.WriteString(end); err != nil {
+			return err
+		}
+	}
+
+	if _, err := builder.WriteString("]\n"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p securityPolicyInternal) marshalRego() (string, error) {
+	builder := new(strings.Builder)
+	if _, err := builder.WriteString(fmt.Sprintf("package policy\nallow_all := %v\n", p.AllowAll)); err != nil {
+		return "", err
+	}
+
+	if err := addContainers(builder, p.Containers); err != nil {
+		return "", err
+	}
+
+	return builder.String(), nil
+}
+
+func createRegoEnforcer(state SecurityPolicyState, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (SecurityPolicyEnforcer, error) {
+	policy, err := state.SecurityPolicy.toInternal()
+	if err != nil {
+		return nil, fmt.Errorf("error converting to internal format: %w", err)
+	}
+
+	regoPolicy, err := newRegoPolicyFromInternal(policy, defaultMounts, privilegedMounts)
+	if err != nil {
+		return nil, fmt.Errorf("error converting to Rego: %w", err)
+	}
+
+	regoPolicy.base64policy = state.EncodedSecurityPolicy.SecurityPolicy
+	return regoPolicy, nil
+}
+
+func newRegoPolicyFromInternal(securityPolicy *securityPolicyInternal, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (*regoEnforcer, error) {
+	policy := new(regoEnforcer)
+	policy.behavior = policyCode
+	var err error
+	policy.objects, err = securityPolicy.marshalRego()
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert json to rego: %w", err)
+	}
+
+	defaultMountData := make([]interface{}, 0, len(defaultMounts))
+	privilegedMountData := make([]interface{}, 0, len(privilegedMounts))
+	policy.data = map[string]interface{}{
```

Comment:
We should have a comment somewhere (maybe here) describing what each of the data fields is used for.